modules/gapi/include/opencv2/gapi/imgproc.hpp

```diff
@@ -102,6 +109,20 @@ namespace imgproc {
         }
     };
 
+    G_TYPED_KERNEL(GCalcOptFlowLK, <GArrOutLK(GMat,GMat,GArrayPoint2f,GArrayPoint2f,cv::Size,int,cv::TermCriteria,int,double)>,
+                   "org.opencv.imgproc.optFlowLK") {
+        static std::tuple<GArrayDesc, GArrayDesc, GArrayDesc> outMeta(GMatDesc, GMatDesc, GArrayDesc, GArrayDesc, const cv::Size&, int, const cv::TermCriteria&, int, double) {
+            return std::make_tuple(empty_array_desc(), empty_array_desc(), empty_array_desc());
+        }
+    };
+
+    G_TYPED_KERNEL(GCalcOptFlowPyrLK, <GArrOutLK(GArrayMat,GArrayMat,GArrayPoint2f,GArrayPoint2f,cv::Size,int,cv::TermCriteria,int,double)>,
+                   "org.opencv.imgproc.optFlowPyrLK") {
+        static std::tuple<GArrayDesc, GArrayDesc, GArrayDesc> outMeta(GArrayDesc, GArrayDesc, GArrayDesc, GArrayDesc, const cv::Size&, int, const cv::TermCriteria&, int, double) {
+            return std::make_tuple(empty_array_desc(), empty_array_desc(), empty_array_desc());
+        }
+    };
```

Comment:
There are two different kernels (almost the same except input GMats/GArrays< GMat >) because cv::calcOptFlow doesn't call pyramid building if we pass to it vector< Mat >, so we can't just invent casting GMat -> GArray< GMat > without functionality lost

Reply:
fixed